### ISSUE TITLE
Unable to install in pypy

### ISSUE THREAD

**Issue by bitnom** [Neoteroi/BlackSheep#314]
To reproduce:
```sh
pyenv install pypy3.9-7.3.11
pyenv local pypy3.9-7.3.11
python -m venv test_pypy
source test_pypy/bin/activate
pip install blacksheep
```

Error:
```
blacksheep/contents.c: In function ‘__Pyx_async_gen_init_hooks’:
      blacksheep/contents.c:22155:23: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘async_gen_finalizer’
      22155 |     finalizer = tstate->async_gen_finalizer;
            |                       ^~
      blacksheep/contents.c:22160:23: error: ‘PyThreadState’ {aka ‘struct _ts’} has no member named ‘async_gen_firstiter’
      22160 |     firstiter = tstate->async_gen_firstiter;
            |                       ^~
      error: command '/usr/bin/gcc' failed with exit code 1
      [end of output]
```

I figured it might pretty awesome if I could use blacksheep in pypy.

**Comment by RobertoPrevato** [comment]
Hi @bitnom 
I am aware of this, in the `README` I also listed the supported runtimes and pypy is intentionally not mentioned. `blacksheep` uses `Cython`, which does not play well along with `PyPy` (also `uvloop` does not support `pypy`).

In the past I took into consideration maintaining also a pure Python implementation of BlackSheep, but I don't have time for this right now. If I had time, I would also try replacing Cython with Rust and maturin, it seems that would give support for pypy, too.
